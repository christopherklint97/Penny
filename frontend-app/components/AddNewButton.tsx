import Link from 'next/link'

export default function AddNewButton({ link, text }) {
  return (
    <Link href={link}>
      <div className="flex flex-row justify-center w-24 p-1 mx-auto text-gray-900 bg-blue-300 rounded-md shadow-lg cursor-pointer hover:bg-blue-400">
        <svg
          className="my-auto mr-1 text-gray-900 fill-current"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          fill="black"
          width="18px"
          height="18px"
        >
          <path d="M0 0h24v24H0z" fill="none" />
          <path d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z" />
        </svg>
        <h2 className="my-auto">{text}</h2>
      </div>
    </Link>
  )
}
