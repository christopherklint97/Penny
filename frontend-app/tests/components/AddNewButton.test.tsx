import React from 'react'
import { render } from '../testUtils'
import AddNewButton from '../../components/AddNewButton'

describe('AddNewButton component', () => {
  it('renders without crashing', function () {
    render(<AddNewButton link="www.example.com" text="test" />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <AddNewButton link="www.example.com" text="test" />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
