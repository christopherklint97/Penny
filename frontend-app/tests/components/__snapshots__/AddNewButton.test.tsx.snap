// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewButton component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row justify-center w-24 mx-auto shadow-lg rounded-md p-1 cursor-pointer bg-blue-300 hover:bg-blue-400 text-gray-900"
  >
    <svg
      class="fill-current text-gray-900 my-auto mr-1"
      fill="black"
      height="18px"
      viewBox="0 0 24 24"
      width="18px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    <h2
      class="my-auto"
    >
      test
    </h2>
  </div>
</DocumentFragment>
`;
